protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,345 @@
+   They cannot use the un-backfilled commits in `_delta_log/_commits` directory as they are not aware of the external commit-owner. Because of this, they may see a stale snapshot
+   of the table if the recent commits are not backfilled.
+    - In the above example, such delta implementation will see version 7 as the latest snapshot.
+2. A client that understands managed commits can supplement the file listing result by contacting the commit-owner for information about un-backfilled commits, in order to get the most
+   recent snapshot of the table.
+    - In the above example, a delta implementation could get information about versions 0 through 7 from _delta_log directory and get information about un-backfilled commits (v8, v9) from the commit-owner.
+
+## Maintenance operations on managed-commit tables
+
+[Checkpoints](#checkpoints-1) and [log compaction files](#log-compaction-files) can only be created for commits in the `_delta_log` directory. In other words, in order to
+checkpoint version `v` or produce a compacted log file for commit range x <= v <= y, `_delta_log/<v>.json` must exist. Otherwise, filesystem-based readers who encountered
+the seemingly-extra files might think the table metadata was corrupted.
+
+## Managed Commit Enablement
+
+The managed-commit feature is supported and active when:
+- The table must be on Writer Version 7.
+- The table has a `protocol` action with `writerFeatures` containing the feature `managedCommits`.
+- The table has a metadata property `delta.managedCommits.commitOwner` in the [change-metadata](#change-metadata)'s configuration.
+- The table may have a metadata property `delta.managedCommits.commitOwnerConf` in the [change-metadata](#change-metadata)'s configuration. The value of this property is a json-coded string-to-string map.
+    - A commit-owner can store additional information (e.g. configuration information such as service endpoints) in this field, for use by the commit-owner client (it is opaque to the Delta client).
+    - This field should never include secrets such as auth tokens or credentials, because any reader with access to the table's storage location can see them.
+
+Note that a table is in invalid state if the change-metadata contains the `delta.managedCommits.commitOwner` property but the table does not have the `managedCommits` feature in the `protocol` action or vice versa.
+
+E.g.
+```json
+{
+  "metaData":{
+    "id":"af23c9d7-fff1-4a5a-a2c8-55c59bd782aa",
+    "format":{"provider":"parquet","options":{}},
+    "schemaString":"...",
+    "partitionColumns":[],
+    "configuration":{
+      "appendOnly": "true",
+      "delta.managedCommits.commitOwner": "commit-owner-1",
+       "delta.managedCommits.commitOwnerConf": "{\"endpoint\":\"http://sample-url.com/commit\", \"authenticationMode\":\"oauth2\"}"
+    }
+  }
+}
+```
+
+## Writer Requirements for Managed Commits
+
+When supported and active:
+- The `inCommitTimestamp` table feature must also be supported and active.
```

Comment:
I don't think this is explained anywhere... probably worth updating the backfill section to point out that ICT is needed because the backfilled commit will otherwise have a different timestamp than the original?

Reply:
Adding following text in `Commit Backfills` section:

Since commit backfills may happen at a later point in time, so the `file modification timestamp` of the backfilled file might be very different than the time of actual commit. For this reason, the `managed-commit` feature depends on another writer feature called [in-commit-timestamps](#TODO-Put-Relevant-Link) to make the commit timestamps more reliable. Refer to [Writer Requirements for Managed Commits](#writer-requirements-for-managed-commits) section for more details.